### PR TITLE
[CLOUD-3454] - EAP 7.3.0 CR1 build configs, imagestreams and templates

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/CLOUD-3454
Signed-off-by: Ken Wills <kwills@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by luck3y** [comment]
Note that these builds are all generated from the same repo now using specific overrides files. As we add other builds (s390x etc) we'll just add specific overrides for those as needed.

File: dev-jdk8-overrides.yaml [hunk 1]
```diff
@@ -29,14 +30,24 @@ modules:
...
+#artifacts:
+#  - name: maven-repo
+#    target: maven-repo.zip
+#    md5: 1f7a658f75d747b40c157f22e26f24a7
 
 osbs:
       repository:
             name: containers/jboss-eap-7
-            branch: jb-eap-7.3-openshift-rhel-8
+            branch: jb-eap-7.3-openjdk-8-openshift-dev-rhel-8
```
Comment (jmesnil) [inline dev-jdk8-overrides.yaml:55]: isn't there a mismatch between the `com.redhat.component` component (`jboss-eap-7-eap73-openjdk8-openshift-rhel7-container`), this image being based on `ubi7:7-released`and using the branch ` jb-eap-7.3-openjdk-8-openshift-dev-rhel-8`?

File: docs/README.adoc [hunk 2]
```diff
@@ -16,14 +16,21 @@ toc::[levels=1]
...
+* link:./templates/eap73-openjdk11-amq-s2i.adoc[eap73-openjdk11-amq-s2i]
+* link:./templates/eap73-openjdk11-basic-s2i.adoc[eap73-openjdk11-basic-s2i]
+* link:./templates/eap73-openjdk11-https-s2i.adoc[eap73-openjdk11-https-s2i]
+* link:./templates/eap73-openjdk11-sso-s2i.adoc[eap73-openjdk11-sso-s2i]
+* link:./templates/eap73-openjdk11-third-party-db-s2i.adoc[eap73-openjdk11-third-party-db-s2i]
+* link:./templates/eap73-openjdk11-tx-recovery-s2i.adoc[eap73-openjdk11-tx-recovery-s2i]
+* link:./templates/eap73-sso-s2i.adoc[eap73-sso-s2i]
+* link:./templates/eap73-third-party-db-s2i.adoc[eap73-third-party-db-s2i]
+* link:./templates/eap73-tx-recovery-s2i.adoc[eap73-tx-recovery-s2i]
+* link:./templates/fn.adoc[fn]
```
Comment (jmesnil) [inline docs/README.adoc]: is the footnote (I presume) link on purpose?

**Comment by jfdenise** [comment]
@luck3y , I notice that the dependency on wildfly-cekit-modules is not right. I just released 0.18.2 that is the one to use for 7.3.
I mean this: https://github.com/luck3y/jboss-eap-7-openshift-image/blob/CLOUD-3454/image.yaml#L47

File: dev-jdk8-overrides.yaml (hunk 1 above)
Comment (luck3y) [reply dev-jdk8-overrides.yaml:55]: Yep. but the images are actually build using the rel-* overrides. I'll update this one to fix.

File: docs/README.adoc (hunk 2 above)
Comment (luck3y) [reply docs/README.adoc]: Good question, never noticed that before, I'll have to look.

**Comment by luck3y** [comment]
> @luck3y , I notice that the dependency on wildfly-cekit-modules is not right. I just released 0.18.2 that is the one to use for 7.3.
> I mean this: https://github.com/luck3y/jboss-eap-7-openshift-image/blob/CLOUD-3454/image.yaml#L47

Ok, I'll update this and rebuild with the correct one. At some point we should talk about how to coordinate this like we do with component upgrades etc.

**Comment by luck3y** [comment]
@jmesnil @jfdenise FYI this was updated, and should be good to go now, can you review and I'll merge it?

File: image.yaml [hunk 3]
```diff
@@ -32,19 +32,19 @@ modules:
           - name: cct_module
             git:
                   url: https://github.com/jboss-openshift/cct_module.git
-                  ref: master
+                  ref: 0.39.0
           - name: jboss-eap-modules
             git:
                   url: https://github.com/jboss-container-images/jboss-eap-modules.git
-                  ref: master
+                  ref: EAP_730_CR1
```
Comment (jfdenise) [inline image.yaml]: Just a note, some commits have been applied to jboss-eap-modules since this tag. I suppose this will be handled in another commit.

**Review by jfdenise — APPROVED** [review]

File: docs/templates/eap73-openjdk11-openjdk11-amq-persistent-s2i.adoc [hunk 4]
```diff
@@ -0,0 +1,387 @@
+////
+    AUTOGENERATED FILE - this file was generated via ./gen_template_docs.py.
+    Changes to .adoc or HTML files may be overwritten! Please change the
+    generator or the input template (./*.in)
+////
+
+= eap73-amq-persistent-s2i
```
Comment (jmesnil) [inline docs/templates/eap73-openjdk11-openjdk11-amq-persistent-s2i.adoc]: * why does `openjdk11` appear twice in the file name?
* shouldn't the title of the file match its filename?

File: docs/templates/eap73-openjdk11-openjdk11-amq-s2i.adoc [hunk 5]
```diff
@@ -0,0 +1,369 @@
+////
+    AUTOGENERATED FILE - this file was generated via ./gen_template_docs.py.
+    Changes to .adoc or HTML files may be overwritten! Please change the
+    generator or the input template (./*.in)
+////
+
+= eap73-amq-s2i
```
Comment (jmesnil) [inline docs/templates/eap73-openjdk11-openjdk11-amq-s2i.adoc]: same remarks than above

File: docs/templates/eap73-openjdk11-openjdk11-basic-s2i.adoc [hunk 6]
```diff
@@ -0,0 +1,289 @@
+////
+    AUTOGENERATED FILE - this file was generated via ./gen_template_docs.py.
+    Changes to .adoc or HTML files may be overwritten! Please change the
+    generator or the input template (./*.in)
+////
+
+= eap73-basic-s2i
```
Comment (jmesnil) [inline docs/templates/eap73-openjdk11-openjdk11-basic-s2i.adoc]: same remarks than above

File: docs/templates/eap73-openjdk11-openjdk11-https-s2i.adoc [hunk 7]
```diff
@@ -0,0 +1,322 @@
+////
+    AUTOGENERATED FILE - this file was generated via ./gen_template_docs.py.
+    Changes to .adoc or HTML files may be overwritten! Please change the
+    generator or the input template (./*.in)
+////
+
+= eap73-https-s2i
```
Comment (jmesnil) [inline docs/templates/eap73-openjdk11-openjdk11-https-s2i.adoc]: same remarks than above

File: image.yaml [hunk 8]
```diff
@@ -105,14 +105,22 @@ modules:
...
+
+# note that building in OSBS using ODCS requires a content set to be specified, these are done in the overrides
+# cekit doesn't appear to allow this to be overriden from a default
+#packages:
+#      content_sets_file: content_sets.yml
+
+artifacts:
+  - name: maven-repo
+    target: maven-repo.zip
+    md5: 927cf0a61a4dccab4fb835d6a5fec2f0
```
Comment (jmesnil) [inline image.yaml:122]: how can I check that the checksum correspond to the correct Maven repository for EAP 7.3CR2?

File: templates/eap73-image-stream.json [hunk 9]
```diff
@@ -0,0 +1,177 @@
+{
+    "kind": "List",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "jboss73-image-streams",
```
Comment (jmesnil) [inline templates/eap73-image-stream.json:5]: File is named `eap73-image-stream` while resource name is `jboss73-image-streams`. Shouldn't they match?

Comment (jmesnil) [reply templates/eap73-image-stream.json:5]: It's not appearing in the file name or resource name but this image stream is for JDK8. Should this be made explicit?
Do we need an image stream for JDK11 too?

File: docs/templates/eap73-openjdk11-openjdk11-amq-persistent-s2i.adoc (hunk 4 above)
Comment (luck3y) [reply docs/templates/eap73-openjdk11-openjdk11-amq-persistent-s2i.adoc]: I'll check the docs generation script, as far as I know we haven't changed anything there, but it does prefix the image name into the file. The filename should match though, yes.

File: image.yaml (hunk 8 above)
Comment (luck3y) [reply image.yaml:122]: The only way to do this is to compare what is here with the checksums created when the maven zip is built by prod. I'll link you the internal URL seperately.

File: image.yaml (hunk 3 above)
Comment (luck3y) [reply image.yaml]: This was added / updated for CR2

File: docs/templates/eap73-openjdk11-openjdk11-amq-persistent-s2i.adoc (hunk 4 above)
Comment (luck3y) [reply docs/templates/eap73-openjdk11-openjdk11-amq-persistent-s2i.adoc]: This was an error, I've removed the bad doc files and regenerated the adoc files, so this should be ok now.

File: templates/eap73-image-stream.json (hunk 9 above)
Comment (luck3y) [reply templates/eap73-image-stream.json:5]: The convention in the past has been to pluralize the list name in the file:

https://github.com/jboss-container-images/jboss-eap-7-openshift-image/blob/7.2.x/templates/eap72-image-stream.json#L5

The JDK 8 / RHEL 7 templates and imagestreams don't container -openjdk8 as part of the name, as that was the convention for 7.2, and what the docs specify.

They currently both us -imagestreams in the names.

There is already an 7.3 OpenJDK 11 version (templates/eap73-openjdk11-image-stream.json).

Comment (luck3y) [reply templates/eap73-image-stream.json:5]: There is already an imagestream for JDK 11 (templates/eap73-openjdk11-image-stream.json)

The naming of the imagestream list within the file matches what we had done previously and is usually pluralized. (https://github.com/jboss-container-images/jboss-eap-7-openshift-image/blob/7.2.x/templates/eap72-image-stream.json#L5)

**Comment by luck3y** [comment]
This includes https://github.com/luck3y/jboss-eap-7-openshift-image/pull/3

**Comment by jmesnil** [comment]
@luck3y thanks
